pkg/operator/ceph/csi/csi.go

```diff
@@ -28,8 +28,12 @@ import (
 )
 
 func ValidateAndConfigureDrivers(context *clusterd.Context, namespace, rookImage, securityAccount string, serverVersion *version.Info, ownerInfo *k8sutil.OwnerInfo) {
-	var v *CephCSIVersion
-	var err error
+	csiLock.Lock()
+	defer csiLock.Unlock()
```

Comment:
Locking the whole function, since disabling both drivers(`CSIEnabled()==false`) may lead directly to calling `stopDrivers` at the end when maybe same function is in another go-routine trying to start drivers.